arviz/plots/kdeplot.py

```diff
@@ -261,6 +264,33 @@ def plot_kde(
         gridsize = (128, 128) if contour else (256, 256)
         density, xmin, xmax, ymin, ymax = _fast_kde_2d(values, values2, gridsize=gridsize)
 
+        if hpd_levels is not None:
+            # Check hpd levels are within bounds [0, 1]
+            if min(hpd_levels) < 0 or max(hpd_levels) > 1:
+                raise ValueError(
+                    "Highest posterior density confidence levels must be between 0 and 1"
+                )
+
+            # Need to include 0 & 1 if not included so that contours plot as expected
+            if 0 not in hpd_levels:
+                hpd_levels.append(0)
+            if 1 not in hpd_levels:
+                hpd_levels.append(1)
```

Comment:
I don't think this should be necessary.

Reply:
The reason I included 0 and 1 was to ensure that all the contours are filled in correctly (`plt.contourf` will not fill the highest contour otherwise, and using the `fill_last` argument to ensure the lowest contour is filled introduces an extraneous contour on the plot that I think outlines all non-zero values)

I've rewritten this section so that it's hopefully clearer that we're including 0 and `density.max()`, but please let me know if it's not clear and I can rewrite it again (or put some comments explaining the reasoning)

Reply:
I think this is clear